src/kvstore.h

```diff
@@ -0,0 +1,81 @@
+#ifndef DICTARRAY_H_
+#define DICTARRAY_H_
+
+#include "dict.h"
+#include "adlist.h"
+
+typedef struct {
+    list *rehashing;                       /* List of dictionaries in this kvstore that are currently rehashing. */
+    int resize_cursor;                     /* Cron job uses this cursor to gradually resize dictionaries. */
+    int non_empty_dicts;                   /* The number of non-empty dicts. */
+    unsigned long long key_count;          /* Total number of keys in this kvstore. */
+    unsigned long long bucket_count;       /* Total number of buckets in this kvstore across dictionaries. */
+    unsigned long long *dict_size_index;   /* Binary indexed tree (BIT) that describes cumulative key frequencies up until given dict-index. */
+} kvstoreState;
```

Comment:
Does this need to be a separate struct? It seems to just add a lot of "->state." everywhere, which doesn't seem that useful.

Reply:
good idea, i'll add it in a seperate commit